solidity_contracts/src/CairoPrecompiles/PragmaCaller.sol

```diff
@@ -7,10 +7,43 @@ using CairoLib for uint256;
 
 contract PragmaCaller {
     /// @dev The starknet address of the pragma oracle
-    uint256 pragmaOracle;
+    uint256 private immutable pragmaOracle;
 
-    /// @dev The cairo function selector to call - `get_data_median`
-    uint256 constant FUNCTION_SELECTOR_GET_DATA_MEDIAN = uint256(keccak256("get_data_median")) % 2 ** 250;
+    /// @dev The starknet address of the pragma summary stats
+    uint256 private immutable pragmaSummaryStats;
```

Comment:
immutables should be put after the constants

Reply:
Also should be renamed `pragmaSummaryStatsAddress` for explicitness

Reply:
The other contracts are naming the address variables without the `Address` suffix. Shall I rename it everywhere? imo it's fine but I can

Reply:
Indeed, there is no Address suffix on other contracts. Let's keep it that way then.
After reflexion, it might not be necessary to add it as the cairoLib functiions called on pragmaSummaryStats are explicitly suggesting the variable is a starknet address